test/cpp/end2end/client_lb_end2end_test.cc

```diff
@@ -539,6 +539,33 @@ TEST_F(ClientLbEnd2endTest, RoundRobin) {
   EXPECT_EQ("round_robin", channel->GetLoadBalancingPolicyName());
 }
 
+TEST_F(ClientLbEnd2endTest, RoundRobinProcessPending) {
+  StartServers(1);  // Single server
+  auto channel = BuildChannel("round_robin");
+  auto stub = BuildStub(channel);
+  SetNextResolution({servers_[0]->port_});
+  WaitForServer(stub, 0, DEBUG_LOCATION);
+  constexpr int kIterations = 4;
+  constexpr int kNumThreads = 4;
+  std::vector<std::thread> threads;
+  // Create and destroy several channels concurrently, executing an RPC each
```

Comment:
We can probably trigger this without multiple threads once #15841 is merged.  Maybe add a TODO assigned to @AspirinSJL?

Reply:
If we don't need to recycle the subchannel to trigger the issue, wouldn't it be sufficient to start one client, wait until its subchannel is READY, then start a second client with the same target? The second client will have a born READY subchannel in this case, right?

Reply:
Probably. However, the issue is time-sensitive, so using threads and a loop makes it much more likely to happen. For testing, the more deterministic the failure, the better.